src/material-experimental/mdc-list/list-option.ts

```diff
@@ -233,5 +236,68 @@ export class MatListOption extends MatListItemBase implements OnInit, OnDestroy
     this._changeDetectorRef.markForCheck();
   }
 
+  /**
+   * Renders the checkbox and icons/avatars based on the set checkbox position. This is needed
+   * because in Angular w/ View Engine, it is not possible to conditionally render a single
+   * embedded view through different containers which change based on a binding. Note that
+   * only one container is active at a time. View Engine throws when an embedded view moves
+   * to another container. See: https://stackblitz.com/edit/angular-ivy-7veqgj
+   */
+  // * TODO: Consider removing if View Engine is no longer supported.
+  private _renderCheckboxAndIcons() {
```

Comment:
I'm sad about the amount of additional code needed to work around this silly ViewEngine issue, but unfortunately I can't really think of a better way

Reply:
I spent more time looking into this (given the new changes for the icons/avatars) and was able to solve the issue in an Angular idiomatic way. Fortunately this block of code is no longer needed!

Initially I was under the impression it is an issue with the embedded view references (and their removal in View Engine), but it looks like View Engine just breaks due to the conditional use of multiple view refs in the same template outlet. The fix now is to just use two outlets (which make sense anyway now due to the new changes for icons/avatars).

StackBlitz showing the issue: https://stackblitz.com/edit/view-engine-container-template-outlet-issue
StackBlitz with the fix/workaround: https://stackblitz.com/edit/view-engine-workaround-template-outlet